bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BObjectType.java

```diff
@@ -136,8 +137,30 @@ public void setTypeIdSet(BTypeIdSet typeIdSet) {
         this.typeIdSet = typeIdSet;
     }
 
-    public void processObjectCtorAnnots(MapValue globalAnnotationMap, Strand strand) {
-        AnnotationUtils.processObjectCtorAnnotations(globalAnnotationMap, this, strand);
+    public BObjectType duplicateTypeAndprocessObjectCtorAnnots(MapValue globalAnnotationMap, Strand strand) {
+        BObjectType clone = cloneThis();
+        AnnotationUtils.processObjectCtorAnnotations(globalAnnotationMap, clone, strand);
+        return clone;
+    }
+
+    BObjectType cloneThis() {
```

Comment:
Doesn't `cloneThis` and `duplicate` do pretty much the same thing? Shouldn't we use one of the two as the name?

Reply:
Will change this to `duplicate`,  this function and the `duplicate` function from function type have nothing common, just that they both duplicate their respective objects.